File: feb-27-minimum-changes-to-make-alternating-binary-string/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var minOperations = function(s) {
  let mismatchStartZero = 0;

  for (let i = 0; i < s.length; i++) {
    const expected = (i % 2 === 0) ? 48 : 49;
    if (s.charCodeAt(i) !== expected) {
      mismatchStartZero++;
    }
  }

  return Math.min(mismatchStartZero, s.length - mismatchStartZero);
};

File: feb-27-minimum-changes-to-make-alternating-binary-string/solution.py
class Solution:
    def minOperations(self, s: str) -> int:
        mismatch_start_zero = 0

        for i, ch in enumerate(s):
            expected = "0" if i % 2 == 0 else "1"
            if ch != expected:
                mismatch_start_zero += 1

        return min(mismatch_start_zero, len(s) - mismatch_start_zero)
